elytron/WFCORE-5120-client-default-ssl-context.adoc

```diff
@@ -0,0 +1,117 @@
+This feature seeks to expose this default SSL context for users without having to interact with Elytron API directly in the code. This will be made possible by implementing a new https://docs.oracle.com/javase/9/docs/api/java/security/Provider.html[java security provider] . This java security provider will provide SSLContext with name "Default", which means that if this provider is registered with high enough priority, it will instantiate and return the configured SSLContext when end users call https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/SSLContext.html#getDefault()[SSLContext.getDefault()] in their code. This way client libraries can make use of it without modifying their code to use Elytron client. They must only register the provider in the JVM either statically by adding of this provider to java security file, or dynamically in the code by using standard java provider registration API.
+
+Changes in code will be done only in Eytron repository.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/ELY-2112[ELY-2112]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFCORE-4144[WFCORE-4144]
+* https://issues.redhat.com/browse/WFLY-13762[WFLY-13762]
+* https://issues.redhat.com/browse/WFCORE-5101[WFCORE-5101]
+* https://issues.redhat.com/browse/WFCORE-5120[WFCORE-5120]
+* https://issues.redhat.com/browse/WFLY-14846[WFLY-14846]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolak Krepinska]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Elytron - security provider and tests will be added
+* WildFly - community documentation will be added
+
+=== Other Interested Projects
+
+NONE
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* Implementation of new https://docs.oracle.com/javase/9/docs/api/java/security/Provider.html[java security provider] that provides SSLContext under algorithm named "Default".
+
+* This provider will be able to load Elytron client configuration and will provide initialized SSL context for any client requesting it by using `SSLContext.getDefault()`.
+
+* The provider will load the Elytron's SSL context configuration either from the current authentication context obtained from classpath, or from authentication context generated by a file that is passed into the security provider (either programmatically to provider's constructor or statically as an argument in a java.security file).
```

Comment:
@Skyllarr  `statically as an argument in a java.security file` Does it still work on JDK11. Weren't  we discussing in FIPS context this feature of JDK does not work anymore? Or it was related just to FIPS context?

Reply:
@mchoma The issue on JDK11 with FIPS was about specifically PKCS11 provider ignoring additonal parameter so therefore it cannot delegate to FIPS provider. but specifying providers in `java.security` file works in JDK11 normally.

Reply:
Thanks for clarifying.